### PR TITLE
drivers: enhance lsm6dsl 

### PR DESCRIPTION
**Pull request by smlng**
- fix temperature conversion
- general optimisation
- add SAUL support for temperature

Test output without this fix:

```
Accelerometer x: -85 y: -24 z: 1034
Temperature value: -777 degrees
Gyroscope x: 10 y: -22 z: -10
```

and with the fixes of this PR:

```
Accelerometer x: -87 y: -26 z: 1034
Temperature value: 2168 degrees
Gyroscope x: 10 y: -21 z: -10
```

The temperature now correctly shows ambient temperature in degree Celsius (with factor x100).

Further this tests reduces binary size by nearly 1.5K

sizes with current master:

```
text	   data	    bss	    dec	    hex	filename
  12900	    136	   2784	  15820	   3dcc	/RIOT/tests/driver_lsm6dsl/bin/nucleo-f401/tests_driver_lsm6dsl.elf
```

sizes with this PR:

```
text	   data	    bss	    dec	    hex	filename
 11556	    136	   2784	  14476	   388c	/RIOT/tests/driver_lsm6dsl/bin/nucleo-f401/tests_driver_lsm6dsl.elf
```

### CODE REVIEW

**Comment by smlng** [comment]
also see #7282

File: drivers/lsm6dsl/include/lsm6dsl_internal.h [hunk 1]
```diff
@@ -149,6 +151,16 @@ extern "C" {
 #define LSM6DSL_FIFO_CTRL3_GYRO_DEC_SHIFT  (3)
 /** @} */
 
+/**
+ * @brief	Offset for temperature calculation
+ */
+#define LSM6DSL_TEMP_OFFSET					(0x1900)
```
Comment (vincent-d) [inline drivers/lsm6dsl/include/lsm6dsl_internal.h]: I would align with the other values

**Comment by aabadie** [comment]
Since the driver returns the temperature in centi-degrees, the output of the test application should also be updated:
```
Temperature value: 2168 **centi-**degrees
```

**Comment by smlng** [comment]
@aabadie IMHO there is no such thing as _centi-degrees_, further the acc and gyro reading have no unit at all, and its a driver test if one wants to nice units use SAUL 😉

**Comment by aabadie** [comment]
My comment is just related to the description of this PR where 2168 degrees sounds a bit irrealistic :)

**Comment by smlng** [comment]
@aabadie I updated the docu of the function to note the scaling factor.

**Comment by aabadie** [comment]
> I updated the docu of the function to note the scaling factor.

Ok, but that doesn't prevent the fact that the `main.c` of the test application should be updated as well ;)

**Comment by miri64** [comment]
Isn't that fixed with #7250 anyways?

**Comment by smlng** [comment]
> Isn't that fixed with #7250 anyways?

nope this is about the plain driver test, no SAUL involved here.

**Comment by smlng** [comment]
@aabadie addressed comments; I guess?

**Comment by vincent-d** [comment]
ACK

**Comment by aabadie** [comment]
and go!

**Comment by aabadie** [comment]
I'm sure if it needs backport. I would say yes because it fixes the driver temperature reading but it also does a lot more...

**Comment by smlng** [comment]
IIRC we mainly backport security and severe bug fixes, a suboptimal sensor reading is not that important - which by the way is not the major reading of this sensor.

**Pull request merged**